google/cloud/aiplatform/models.py

```diff
@@ -852,7 +854,42 @@ def predict(self, instances: List, parameters: Optional[Dict] = None) -> Predict
     # TODO(b/172828587): implement prediction
     def explain(self, instances: List[Dict], parameters: Optional[Dict]) -> List[Dict]:
         """Online prediction with explanation."""
-        raise NotImplementedError("Prediction not implemented.")
+        raise NotImplementedError("Prediction with explanation not implemented.")
+
+    def list_models(self) -> Sequence[gca_endpoint.DeployedModel]:
+        """Returns a list of the models deployed to this Endpoint.
+
+        Returns:
+            deployed_models (Sequence[aiplatform.gapic.DeployedModel]):
+                A list of the models deployed in this Endpoint.
+        """
+        self._sync_gca_resource()
+        return self._gca_resource.deployed_models
+
+    def undeploy_all(self):
+        """Undeploys every model deployed to this Endpoint."""
+        self._sync_gca_resource()
+
+        for deployed_model in self._gca_resource.deployed_models:
+            self.undeploy(deployed_model_id=deployed_model.id)
+
+    def delete_endpoint(self, force: bool = False):
```

Comment:
To make this optional `sync` we can add a sync argument and forward it to `undeploy_all`. The delete endpoint call should be a private method `_delete` with `@base.optional_sync` and this method should call that private method and forward the `sync` flag.